settlement-pipelines/src/bin/init_settlement.rs

```diff
@@ -0,0 +1,747 @@
+                .as_ref()
+                .map_or(settlement_record.max_total_claim, |settlement| {
+                    settlement
+                        .max_total_claim
+                        .saturating_sub(settlement.lamports_funded)
+                }),
+        };
+
+        if amount_to_fund == 0 {
+            info!(
+                "Settlement account {}/{:?} already funded, skipping funding",
+                settlement_record.settlement_address, settlement_record.funder
+            );
+            settlement_record.state = SettlementRecordState::AlreadyFunded;
+            continue;
+        }
+
+        match settlement_record.funder {
+            SettlementFunderType::Marinade(_) => {
+                info!(
+                    "Settlement account {} is to be funded by Marinade from fee wallet",
+                    settlement_record.settlement_address
+                );
+                settlement_record.funder =
+                    SettlementFunderType::Marinade(Some(SettlementFunderMarinade {
+                        amount_to_fund,
+                    }));
+            }
+            SettlementFunderType::ValidatorBond(_) => {
+                info!(
+                    "Settlement account {} is to be funded by validator from bond stake accounts",
+                    settlement_record.settlement_address
+                );
+
+                let mut empty_vec: Vec<FundBondStakeAccount> = vec![];
+                let funding_stake_accounts = fund_bond_stake_accounts
+                    .get_mut(&settlement_record.vote_account_address)
+                    .unwrap_or(&mut empty_vec);
+                let mut lamports_available: u64 = 0;
+                let mut stake_accounts_to_fund: Vec<FundBondStakeAccount> = vec![];
+                funding_stake_accounts.retain(|stake_account| {
+                    if lamports_available < amount_to_fund + minimal_stake_lamports {
+                        lamports_available += stake_account.lamports;
+                        stake_accounts_to_fund.push(stake_account.clone());
+                        true // delete from the list, no awailable anymore, it will be funded
+                    } else {
+                        false // do not delete, it can be used for other settlement
+                    }
+                });
+
+                let stake_account_to_fund =
+                    if stake_accounts_to_fund.len() == 0 || lamports_available == 0 {
+                        None
+                    } else {
+                        Some(stake_accounts_to_fund.remove(0))
+                    };
+                if let Some(FundBondStakeAccount {
+                    stake_account: destination_stake,
+                    split_stake_account: destination_split_stake,
+                    ..
+                }) = stake_account_to_fund
+                {
+                    info!(
+                        "Settlement:{}, merging stake accounts to {}",
```

Comment:
Merging seems unnecessary but let's keep it here

Reply:
Hm... maybe I was a bit overthinking. Maybe the merge process would be better to be in part of the cleaning pipeline where rent is gathered or similar. Or merging belongs rather to part of claiming the most probably.

My thinking was to have prepared stake account that's prepared for any claiming and the claiming pipeline does not need to be interested in. But  that's not right. It should be the claim that merges when necessary.
This pipeline should be without merge and thus simpler. Considered for refactor.